chainbase/src/main/java/org/tron/core/service/RewardViCalService.java

```diff
@@ -99,25 +98,30 @@ private boolean isDone() {
   }
 
   private void maybeRun() {
-    if (enableNewRewardAlgorithm()) {
-      if (this.newRewardCalStartCycle > 1) {
-        if (isDone()) {
-          this.clearUp(true);
-          logger.info("rewardViCalService is already done");
-        } else {
-          if (lastBlockNumber ==  Long.MAX_VALUE // start rewardViCalService immediately
-              || this.getLatestBlockHeaderNumber() > lastBlockNumber) {
-            // checkpoint is flushed to db, so we can start rewardViCalService
-            startRewardCal();
-            clearUp(true);
+    try {
+      if (enableNewRewardAlgorithm()) {
+        if (this.newRewardCalStartCycle > 1) {
+          if (isDone()) {
+            this.clearUp(true);
+            logger.info("rewardViCalService is already done");
           } else {
-            logger.info("startRewardCal will run after checkpoint is flushed to db");
+            if (lastBlockNumber ==  Long.MAX_VALUE // start rewardViCalService immediately
+                || this.getLatestBlockHeaderNumber() > lastBlockNumber) {
+              // checkpoint is flushed to db, so we can start rewardViCalService
+              startRewardCal();
+              clearUp(true);
+            } else {
+              logger.info("startRewardCal will run after checkpoint is flushed to db");
+            }
           }
+        } else {
+          clearUp(false);
+          logger.info("rewardViCalService is no need to run");
         }
-      } else {
-        clearUp(false);
-        logger.info("rewardViCalService is no need to run");
       }
+    } catch (Exception e) {
```

Comment:
In which case shall it throw an Exception?

Reply:
Such as `too many open files`, `not enough disk space`, etc.